### PR TITLE
fix: MenuItem with icon only will still have additional margin

### LINKED ISSUE #21917: Icon not centered in horizontal menu item
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://codepen.io/AlephOfosev/live/abOLQdN](https://codepen.io/AlephOfosev/live/abOLQdN)

### Steps to reproduce
In a horizontal menu, create a menu item containing only an icon.

The code provided in CodePen is simply the first example provided for Menu component on Ant Design site: https://ant.design/components/menu/
where the text of the first menu item was removed (line 27 removed).

### What is expected?
Icon centered relatively to menu item, notably relatively to bar underlining active menu item when selected.

### What is actually happening?
The icon is not centered, as it has a 10 px margin to its right (see first "mail" icon in the CodePen example).

| Environment | Info |
|---|---|
| antd | 4.0.1 |
| React | 16.13.0 |
| System | Windows 10 |
| Browser | Chrome 79 |

---
The 10px margin-right seems to come from ".ant-menu-item .anticon" selector in css/less.

This makes sense when there is text to its right, but not when there is nothing to the right of the icon.

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #21917

### 💡 Background and solution

css selector can not handle text node. Add additional className for check if only one child.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix MenuItem in horizontal Menu with only one icon still have extra margin.     |
| 🇨🇳 Chinese |    修复 Menu 水平模式下 MenuItem  只有一个 Icon 的时候仍然有额外 margin 的问题。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/menu/style/index.less [hunk 1]
```diff
@@ -203,6 +203,16 @@
         margin-left: 10px;
       }
     }
+
+    &.@{menu-prefix-cls}-item-only-child {
+      .@{iconfont-css-prefix} {
```
Comment (afc163) [inline components/menu/style/index.less]: 能不能用 `.@{iconfont-css-prefix}:only-child` 伪类？

Comment (zombieJ) [reply components/menu/style/index.less]: > css selector can not handle text node. Add additional className for check if only one child.

Comment (zombieJ) [reply components/menu/style/index.less]: `:last-child` `:only-child` 都试过了，text node 都选不出来。

Comment (afc163) [reply components/menu/style/index.less]: 为啥需要选 text node，不是 icon 么。

https://codepen.io/afc163-1472555193/pen/qBdPGXW

Comment (zombieJ) [reply components/menu/style/index.less]: 因为你只选择了 mail 的图标……试试下面这个：

```diff
- .anticon.anticon-mail:only-child {
+ .anticon:only-child {
```

Comment (zombieJ) [reply components/menu/style/index.less]: 截屏2020-03-06上午11 24 22

Comment (afc163) [reply components/menu/style/index.less]:
```css
.ant-menu-item > .anticon:only-child {
  margin-right: 0;
}
```

这样如何

Comment (zombieJ) [reply components/menu/style/index.less]: 外层 selector 是没用的， text node 是选不到的。

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
close #21984 too

**Pull request merged**